src/plugins/fs/mount.c

```diff
@@ -618,7 +618,7 @@ gboolean bd_fs_unmount (const gchar *spec, gboolean lazy, gboolean force, const
                 run_as_uid = g_ascii_strtoull ((*extra_p)->val, NULL, 0);
 
                 /* g_ascii_strtoull returns 0 in case of error */
-                if (run_as_uid == 0 && (g_strcmp0 ((*extra_p)->opt, "0") != 0)) {
+                if (run_as_uid == 0 && (g_strcmp0 ((*extra_p)->val, "0") != 0)) {
```

Comment:
Good catch, however...

This is just a weird test. Conversion error detection should be done by checking the value returned via the second argument of `g_ascii_strtoull()`.

Reply:
Ok, time to do that correctly.